back/engines/free/email_campaigns/app/views/email_campaigns/project_published_mailer/campaign_mail.mjml

```diff
@@ -0,0 +1,30 @@
+<!-- Project preview -->
+<mj-section
+  padding="20px 25px 0"
+  border-radius="5px"
+  text-align="left">
+  <mj-column
+    padding="25px"
+    background-color="#F2F6F8"
+    border-radius="5px">
+    <mj-text>
+      <h2 style="font-size: 18px; margin-top: 10px; margin-bottom: 10px;">
+        <%= localize_for_recipient(event.project_title_multiloc) %>
+      </h2>
+    </mj-text>
+  </mj-column>
+</mj-section>
+
+<mj-section padding="0 0 10px">
+  <mj-column>
+    <mj-button
+      href="<%= event.project_url %>"
+      border-radius="5px"
+      background-color="#044D6C"
+      color="#fff"
+      width="100%"
+      padding="0 25px"><%= format_message('cta_goto_project', component: 'general') %></mj-button>
```

Comment:
I see this `component` is used [in application_mailer](https://github.com/CitizenLabDotCo/citizenlab/blob/1202b3b96dc35cd732e981b1c507fb575eaef245/back/engines/free/email_campaigns/app/mailers/email_campaigns/application_mailer.rb#L40), but I cant find anything with `email_campaigns.general` in the codebase. What am I missing?

Reply:
Ahh, I found it in `back/engines/free/email_campaigns/config/locales/en.yml`
```YAML
general:
      by_author: 'by %{authorName}'
      author_wrote: '%{authorName} wrote:'
      cta_goto_idea: 'Go to this idea'
      cta_goto_proposal: 'Go to this proposal'
      cta_goto_project: 'Go to this project'
```
:+1:

Reply:
I think the name "component" is very confusing, but I reused what was already there. Basically a component doesn't really exist; it's just a part of the path of the translation keys. In this case, translation keys with "general" in the path were introduced. I did this as I noticed we had a lot of duplicate translations, with inconsistent translations, so it seemed valuable to translate them only once so translators need to spend less time translating and for consistency.